.github/workflows/canary-integration-test.yml

```diff
@@ -13,6 +13,8 @@ defaults:
     # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
     shell: bash --noprofile --norc -eo pipefail -x {0}
 
+permissions:
+  contents: read
```

Comment:
nit: could you add a blank line after this?

Reply:
done